folium/features.py

```diff
@@ -197,8 +199,46 @@ def __init__(self, data):
             {% endmacro %}
             """)
 
+class TopoJson(MacroElement):
+    def __init__(self, data, object_path):
+        """TODO docstring here.
+        """
+        super(TopoJson, self).__init__()
+        self._name = 'TopoJson'
+        if 'read' in dir(data):
+            self.data = data.read()
+        elif type(data) is dict:
+            self.data = json.dumps(data)
+        else:
+            self.data = data
+
+        self.object_path = object_path
+
+        self._template = Template(u"""
+            {% macro script(this, kwargs) %}
+                var {{this.get_name()}}_data = {{this.data}};
+                var {{this.get_name()}} = L.geoJson(topojson.feature(
+                    {{this.get_name()}}_data,
+                    {{this.get_name()}}_data.{{this.object_path}}
+                    )).addTo({{this._parent.get_name()}});
+            {% endmacro %}
+            """)
+    def render(self,**kwargs):
+        super(TopoJson,self).render(**kwargs)
+
+        figure = self.get_root()
+        assert isinstance(figure,Figure), ("You cannot render this Element "
+            "if it's not in a Figure.")
+
+        figure.header.add_children(\
+            JavascriptLink("https://cdnjs.cloudflare.com/ajax/libs/topojson/1.6.9/topojson.min.js"),
+                                   name='topojson')
+
 class GeoJsonStyle(MacroElement):
-    def __init__(self, color_domain, color_code, color_data=None, key_on='feature.properties.color'):
+    def __init__(self, color_domain, color_code, color_data=None,
+                 key_on='feature.properties.color',
+                 weight=1, opacity=1, color='black',
+                 fill_opacity=0.6, dash_array=0):
```

Comment:
I think we have to keep `weight, opacity, color, fill_opacity, dash_array` because they are the Leaflet's names.

For the others, I'm open to ideas.

Anyway, I think this will change soon, because today's code enables to do choropleth (have a different `fillColor` per feature), but you can imagine changing another parameter (`color` for example) per feature.

If you're okay, I would like to keep it like this and open another PR to tackle it.